### ISSUE TITLE
Do not import Category

### ISSUE THREAD

**Issue by sebastienserre** [10up/distributor#582]
Hello,

I need to remove categories on pull, which is the best hook to do it?

Thank you

**Comment by dkotter** [comment]
@sebastienserre Thanks for the question!

There are a few ways to go about this, depending on your specific use case.

1. The [`dt_pull_post` action](https://10up.github.io/distributor/dt_pull_post.html) fires after an individual item has been pulled in. You could use this hook to then remove any categories from that item.
2. Using the above hook, categories will still be created and added to items and then you have to remove those after the fact. If you want to prevent categories from being created all together, you can remove those using the [`dt_remote_get` filter](https://10up.github.io/distributor/dt_remote_get.html) or remove the category taxonomy from being synced using the [`dt_syncable_taxonomies` filter](https://10up.github.io/distributor/dt_syncable_taxonomies.html). Just be aware that both of those filters will fire in other contexts as well, so if you go with one of those, make sure there's not unintended consequences

**Comment by sebastienserre** [comment]
Hello @dkotter 
You're right, I've finaly used `dt_pull_post` and it seems OK.

Mny thanks

**Comment by francisco-domo** [comment]
Hello, @sebastienserre sorry for bringing up this old post. I need to do the same thing, remove the category from the pulled posts, could you share the snippet of code you used for achieving this? I'm a bit of a noob and I'm lost regarding what should I do with dt_pull_post.

Thanks in advance.

**Comment by sebastienserre** [comment]
Hello @francisco-domo 

Here's the code I use:
```
add_action( 'dt_pull_post', 'thfo_add_term', 50, 3);
function thfo_add_term( $new_post, $obj, $post_array ) {
	/**
	 * Remove actualité terms
	 */
	$actu_term    = get_term_by( 'slug', 'actualite', 'category' );
	$actu_term_id = $actu_term->term_id;
	$post_terms   = get_the_terms( $new_post, 'category' );
	foreach ( $post_terms as $terms ) {
		if ( $actu_term_id === $terms->term_id ) {
			wp_remove_object_terms( $new_post, $actu_term_id, 'category' );
		}
	}

	/**
	 * Add Term external.
	 * Create it if doesn't exist.
	 */
	$term = get_term_by( 'name', 'external', 'category' );
	if ( ! $term ) {
		$array = wp_insert_term( 'external', 'category' );
	}
	if (is_array( $array ) ){
		$term->id = $array['term_id'];
	}
	$insert = wp_set_post_terms( $new_post, [ $term->term_id ], 'category', true);

	return $insert;
}
```

**Comment by francisco-domo** [comment]
> Hello @francisco-domo
> 
> Here's the code I use:
> 
> ```
> add_action( 'dt_pull_post', 'thfo_add_term', 50, 3);
> function thfo_add_term( $new_post, $obj, $post_array ) {
> 	/**
> 	 * Remove actualité terms
> 	 */
> 	$actu_term    = get_term_by( 'slug', 'actualite', 'category' );
> 	$actu_term_id = $actu_term->term_id;
> 	$post_terms   = get_the_terms( $new_post, 'category' );
> 	foreach ( $post_terms as $terms ) {
> 		if ( $actu_term_id === $terms->term_id ) {
> 			wp_remove_object_terms( $new_post, $actu_term_id, 'category' );
> 		}
> 	}
> 
> 	/**
> 	 * Add Term external.
> 	 * Create it if doesn't exist.
> 	 */
> 	$term = get_term_by( 'name', 'external', 'category' );
> 	if ( ! $term ) {
> 		$array = wp_insert_term( 'external', 'category' );
> 	}
> 	if (is_array( $array ) ){
> 		$term->id = $array['term_id'];
> 	}
> 	$insert = wp_set_post_terms( $new_post, [ $term->term_id ], 'category', true);
> 
> 	return $insert;
> }
> ```

That was fast! Thanks very much.

**Comment by francisco-domo** [comment]
Sorry for this broad question but I'm trying to use this function to remove any category when a post is pulled and let Wordpress assign just the default one but I can't get it to work.  Is it possible to do this?
Thanks for your help.

**Comment by jeffpaul** [comment]
@francisco-domo it looks like the code that @sebastienserre provided above does what you're looking for.  If you're in need of specific engineering help on extending Distributor then email me (jeff.paul@10up.com) and we can discuss options... thanks!

**Comment by sebastienserre** [comment]
Hello,
Sorry @francisco-domo, it seems I've missed some notifications.
I confirm my code above is working great on a customer website.

**Comment by francisco-domo** [comment]
Hello Jeff, Sebastien.

Indeed, the code worked perfectly and it was really helpful, however what I meant to say is that I need to scrap any category that is attached to post (without knowing the name of the category in advance, as in the example where you remove the category "actualite").

I fiddled around with the code and I managed to remove the category but unfortunately it also removed the category of every post I already had in my blog.

I'm not sure if it is possible to do. Thanks very much for your willing to help.
